src/main/java/org/jsmart/zerocode/core/kafka/helper/KafkaConsumerHelper.java

```diff
@@ -0,0 +1,250 @@
+
+            return consumer;
+
+        } catch (IOException e) {
+            throw new RuntimeException("Exception while reading kafka properties and creating a consumer- " + e);
+        }
+    }
+
+    public static void validateConsumeProperties(ConsumerLocalConfigs consumeLocalTestProps) {
+        if (null != consumeLocalTestProps.getFileDumpType() && consumeLocalTestProps.getFileDumpTo() == null) {
+            throw new RuntimeException("Found type, but no fileName. Try e.g. 'fileDumpTo':'target/temp/abc.txt' ");
+        }
+    }
+
+    private static void validateIfBothEnabled(Boolean commitSync, Boolean commitAsync) {
+        if ((commitSync != null && commitAsync != null) && commitSync == true && commitAsync == true) {
+            throw new RuntimeException("\n********* Both commitSync and commitAsync can not be true *********\n");
+        }
+    }
+
+    public static void validateLocalConfigs(ConsumerLocalConfigs consumeLocalTestProps) {
+        if (consumeLocalTestProps != null) {
+            Boolean localCommitSync = consumeLocalTestProps.getCommitSync();
+            Boolean localCommitAsync = consumeLocalTestProps.getCommitAsync();
+
+            validateIfBothEnabled(localCommitSync, localCommitAsync);
+        }
+    }
+
+    public static void validateCommonConfigs(ConsumerCommonConfigs consumerCommonConfigs) {
+        validateIfBothEnabled(consumerCommonConfigs.getCommitSync(), consumerCommonConfigs.getCommitAsync());
+    }
+
+    public static ConsumerLocalConfigs deriveEffectiveConfigs(ConsumerLocalConfigs consumerLocalTestConfigs, ConsumerCommonConfigs consumerCommonConfigs) {
+
+        validateCommonConfigs(consumerCommonConfigs);
+        validateLocalConfigs(consumerLocalTestConfigs);
+
+        return createEffective(consumerCommonConfigs, consumerLocalTestConfigs);
+    }
+
+    public static ConsumerLocalConfigs createEffective(ConsumerCommonConfigs consumerCommon, ConsumerLocalConfigs consumerLocal) {
+        if (consumerLocal == null) {
+            return new ConsumerLocalConfigs(
+                    consumerCommon.getRecordType(),
+                    consumerCommon.getFileDumpTo(),
+                    consumerCommon.getFileDumpType(),
+                    consumerCommon.getCommitAsync(),
+                    consumerCommon.getCommitSync(),
+                    consumerCommon.getShowConsumedRecords(),
+                    consumerCommon.getMaxNoOfRetryPollsOrTimeouts(),
+                    consumerCommon.getPollingTime());
+        }
+
+        // Handle recordType
+        String effectiveRecordType = ofNullable(consumerLocal.getRecordType()).orElse(consumerCommon.getRecordType());
+
+        // Handle fileDumpTo
+        String effectiveFileDumpTo = ofNullable(consumerLocal.getFileDumpTo()).orElse(consumerCommon.getFileDumpTo());
+
+        // Handle fileDumpType
+        String effectiveFileDumpType = ofNullable(consumerLocal.getFileDumpType()).orElse(consumerCommon.getFileDumpType());
```

Comment:
can be removed I think?

Reply:
yes, will handle as part of refactor work.